torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -324,21 +326,22 @@ ComputationClient::DataPtr PjRtComputationClient::ReplicateShardedData(
       // Data is replicated, return the first shard
       return sharded_data->shards[0];
     }
-    xla::XlaBuilder b("ReplicateShardedData");
+    xla::XlaBuilder builder("ReplicateShardedData");
     xla::Shape shape = sharded_data->shape();
-    b.SetSharding(sharded_data->GetSharding());
+    builder.SetSharding(sharded_data->GetSharding());
 
     // perform a simple identity calculation to reassemble the input as
     // replicated output.
-    auto x = xla::Parameter(&b, 0, shape, "p0");
-    b.SetSharding(xla::HloSharding::Replicate().ToProto());
-    xla::XlaOp scalar_two_op =
-        xla::ConvertElementType(xla::ConstantR0(&b, 2), shape.element_type());
-    auto y = xla::Div(x, scalar_two_op);
-    auto z = xla::Add(y, y);
+    xla::XlaOp x = xla::Parameter(&builder, 0, shape, "p0");
+    builder.SetSharding(xla::HloSharding::Replicate().ToProto());
+    xla::XlaOp scalar_zero_op = xla::ConvertElementType(
+        xla::ConstantR0(&builder, 0), shape.element_type());
+    xla::XlaOp y = xla::Add(x, scalar_zero_op);
```

Comment:
Is it possible to have a totally no-op computation and remove the `Add`? Since we're setting `allow_spmd_sharding_propagation_to_output=false`, the result will always be replicated regardless of the input sharding

Reply:
let me try

Reply:
commented out the `xla::add` gives incorrect result. I think there are ways to make it work but since it is not a priority I will leave that to a separate pr.

Reply:
Got it, thanks for trying.